parastell/magnet_coils.py

```diff
@@ -622,49 +619,125 @@ def geometry_file(self, file_path):
         self.coil_solids = []
         for item in imported_geometry:
             if isinstance(item, cq.occ_impl.shapes.Compound):
-                self.coil_solids.extend([[solid] for solid in item.Solids()])
+                self.coil_solids.extend([solid for solid in item.Solids()])
             elif isinstance(item, cq.occ_impl.shapes.Solid):
-                self.coil_solids.append([item])
+                self.coil_solids.append(item)
             else:
                 e = ValueError(
                     f"Imported object of type {type(item)} not recognized."
                 )
                 self._logger.error(e.args[0])
 
-    @property
-    def volume_ids(self):
-        return self._volume_ids
+    def _group_solids(self):
+        """Detects nested solids and groups them together by imported solid ID.
+        (Internal function not intended to be called externally)
+        """
+        centers = [solid.CenterOfBoundBox() for solid in self.coil_solids]
+        center_angles = [np.arctan2(center.y, center.x) for center in centers]
+
+        sorted_angles = np.array(sorted(center_angles))
+        sorted_coils = np.array(
+            [
+                id
+                for _, id in sorted(
+                    zip(center_angles, self.coil_solids),
+                    key=lambda pair: pair[0],
+                )
```

Comment:
In a case of "careful what you ask for 😀 ", now I might want a couple of brief comments here...

Is `id` a member of `self.coil_solids`?  If so, is `id` the best name for this variable?

Or maybe `id` is the index of the entries that you are sorting?